### PR TITLE
pkgsrc fixes

### PR DESCRIPTION
**Pull request by jperkin**
These changes are required for the retro images I use for pkgsrc builds. I've written them such that the current defaults remain intact - both disabling NAT and enabling sendmail are opt-in.

### CODE REVIEW

File: custom/svc/method/smartos-setup [hunk 1]
```diff
@@ -16,28 +17,34 @@
...
-    [[ -n ${ext_ip} ]] || exit $SMF_EXIT_ERR_FATAL
 
     # Don't need Ur
     svcadm disable svc:/smartdc/agent/ur:default || /bin/true
 
     # Don't need mDNS spewing errors
     svcadm disable svc:/network/dns/multicast:default || /bin/true
 
-    # NOPE
-    svcadm disable svc:/network/smtp:sendmail svc:/network/sendmail-client:default
```
Comment (bahamat) [inline custom/svc/method/smartos-setup:40]: What mechanism do you use to get `sendmail=yes` into the config?

**Review by bahamat — APPROVED** [review]

**Review by danmcd — COMMENTED** [review]
Please document how you have tested this, or state that it's a chicken-egg problem where you must push first in order to test it?

**Comment by jperkin** [comment]
I've been running with non-configurable versions of these changes for both 201612 and 202108 builds for the past few years (see the [pkgsrc-retro](https://github.com/TritonDataCenter/joyent-retro/tree/pkgsrc-retro) branch), and have tested this configurable version in a local zone to check the variables work as expected.

I haven't yet tested this in a full pkgsrc configuration, but will do that as soon as there is a built image available in MNX.

**Review by danmcd — APPROVED** [review]
Thank you.

**Pull request merged**